CoursesManager.UI/DataAccess/CourseDataAccess.cs

```diff
@@ -1,30 +1,115 @@
 ﻿using CoursesManager.UI.Models;
 using MySql.Data.MySqlClient;
 using CoursesManager.UI.Database;
+using System.IO;
+using System.Windows.Media.Imaging;
 
 namespace CoursesManager.UI.DataAccess;
 
 public class CourseDataAccess : BaseDataAccess<Course>
 {
     public List<Course> GetAll()
     {
-        throw new NotImplementedException();
+        try
+        {
+            // Voer de stored procedure uit
+            var results = ExecuteProcedure("spCourses_GetAll");
+
+            // Converteer de resultaten naar een lijst van Course-objecten
+            return results.Select(row => new Course
+            {
+                Id = Convert.ToInt32(row["course_id"]),
+                Name = row["course_name"]?.ToString() ?? string.Empty,
+                Code = row["course_code"]?.ToString() ?? string.Empty,
+                Description = row["course_description"]?.ToString() ?? string.Empty,
+                LocationId = Convert.ToInt32(row["course_location_id"]),
+                Location = new Location
+                {
+                    Id = Convert.ToInt32(row["course_location_id"]),
+                    Name = row["location_name"]?.ToString() ?? string.Empty
+                },
+                IsActive = Convert.ToBoolean(row["is_active"]),
+                StartDate = Convert.ToDateTime(row["start_date"]),
+                EndDate = Convert.ToDateTime(row["end_date"]),
+                DateCreated = Convert.ToDateTime(row["created_at"]),
+                Image = row.ContainsKey("tile_image") && row["tile_image"] != DBNull.Value
+                    ? (byte[])row["tile_image"]
+                    : null
+            }).ToList();
+        }
+        catch (Exception ex)
+        {
+            LogUtil.Error($"Error in GetAll: {ex.Message}");
+            throw;
+        }
     }
 
     public void Add(Course course)
     {
-        ArgumentNullException.ThrowIfNull(course);
+        try
+        {
+            ExecuteNonProcedure(
+                StoredProcedures.CourseAddById,
```

Comment:
Kun je nog wat doen aan de naam? Je insert in dit geval niet by id.

Reply:
Ga ik doen!